tower-http/src/services/fs/serve_dir.rs

```diff
@@ -192,6 +203,32 @@ impl<ReqBody> Service<Request<ReqBody>> for ServeDir {
     }
 }
 
+fn parse_range(headers: &HeaderMap) -> Option<ParsedRangeHeader> {
+    // ex:  `Range: bytes=0-1023`
+    let range_header = headers.get(header::RANGE)?;
+    if let Ok(as_str) = range_header.to_str() {
+        // produces ("bytes", "0-1023")
+        return if let Some((_, range)) = as_str.split_once("=") {
+            // produces ("0", "1023")
+            if let Some((start_incl, end_incl)) = range.split_once("-") {
+                if let Ok(start) = start_incl.parse() {
+                    if let Ok(end) = end_incl.parse() {
+                        return Some(ParsedRangeHeader::Valid(ContentRange {
+                            start_inclusive: start,
+                            end_inclusive: end
```

Comment:
Do we check start < end somewhere?

Reply:
Not explicitly, it's done here https://github.com/tower-rs/tower-http/blob/dfd3e04059b9672bbf71521854b040c08ef5cbab/tower-http/src/services/fs/serve_dir.rs#L331 and would panic during the body creation, might not be what we want, I also would like an ending range-value to be an invalid parse, it's a good suggestion to check beforehand!

Reply:
Fixed, but the parsing leaves a lot to be desired and should potentially be reworked

Reply:
:+1: Yeah its nice to avoid panics that can be caused by external clients